python_modules/dagster/dagster_tests/core_tests/resource_tests/pythonic_resources/test_env_vars.py

```diff
@@ -108,10 +262,13 @@ def hello_world_asset(writer: WriterResource):
     try:
         assert (
             defs.get_implicit_global_asset_job_def()
+            # I do not think this should work. Worth discussing in PR.
+            # .execute_in_process(
+            #     {"resources": {"writer": {"config": {"prefix": EnvVar("MY_PREFIX_FOR_TEST")}}}}
+            # )
             .execute_in_process(
-                {"resources": {"writer": {"config": {"prefix": EnvVar("MY_PREFIX_FOR_TEST")}}}}
-            )
-            .success
+                {"resources": {"writer": {"config": {"prefix": {"env": "MY_PREFIX_FOR_TEST"}}}}}
+            ).success
```

Comment:
Undo this to preserve existing behavior, but we should discuss changing this behavior when ben gets back

Reply:
@dpeng817 This is the case we were discussing on the phone